packages/jsii-diff/bin/jsii-diff.ts

```diff
@@ -0,0 +1,131 @@
+import fs = require('fs-extra');
+import reflect = require('jsii-reflect');
+import spec = require('jsii-spec');
+import log4js = require('log4js');
+import yargs = require('yargs');
+import { compareAssemblies } from '../lib';
+import { Mismatches } from '../lib/types';
+import { downloadNpmPackage } from '../lib/util';
+import { VERSION } from '../lib/version';
+
+const LOG = log4js.getLogger('jsii-diff');
+
+async function main(): Promise<number> {
+  const argv = yargs
+      .env('JSII_DIFF')
+      .option('verbose', { alias: 'v', type: 'count', desc: 'Increase the verbosity of output', global: true })
+      .option('default-experimental', { alias: 'e', type: 'boolean', desc: 'Treat unmarked APIs as experimental', default: false })
+      .usage('$0 <original> [updated]', 'Compare two JSII assemblies.', args => args
+        .positional('original', {
+          description: 'Original assembly (file, package or "npm:package@version")',
+          type: 'string'
+        })
+        .positional('updated', {
+          description: 'New assembly (file, package or "npm:package@version")',
+          type: 'string',
+          default: '.'
+        })
+      )
+      .help()
+      .version(VERSION)
+      .argv;
+
+  configureLog4js(argv.verbose);
+
+  LOG.debug(`Loading original assembly from ${(argv as any).original}`);
+  const original = await loadAssembly((argv as any).original);
```

Comment:
I can't get it to work for positional arguments inside a command. I think it's because you're supposed to put the code to run for the command inside a callback inside the `yargs` command.

Reply:
...when people get too fancy with type maps...